test/e2e/metriconly/metrics_test.go

```diff
@@ -57,42 +60,77 @@ var _ = ginkgo.Describe("NPD should export Prometheus metrics.", func() {
 			},
 			*image,
 			*imageProject)
-		if err != nil {
-			ginkgo.Fail(fmt.Sprintf("Unable to create test instance: %v", err))
-		}
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Unable to create test instance: %v", err))
 
 		err = npd.SetupNPD(instance, *npdBuildTar)
-		if err != nil {
-			ginkgo.Fail(fmt.Sprintf("Unable to setup NPD: %v", err))
-		}
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Unable to setup NPD: %v", err))
 	})
 
 	ginkgo.Context("On a clean node", func() {
 
 		ginkgo.It("NPD should export host_uptime metric", func() {
 			err := npd.WaitForNPD(instance, []string{"host_uptime"}, 120)
-			if err != nil {
-				ginkgo.Fail(fmt.Sprintf("Expect NPD to become ready in 120s, but hit error: %v", err))
-			}
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Expect NPD to become ready in 120s, but hit error: %v", err))
 
 			gotMetrics, err := npd.FetchNPDMetrics(instance)
-			if err != nil {
-				ginkgo.Fail(fmt.Sprintf("Error fetching NPD metrics: %v", err))
-			}
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error fetching NPD metrics: %v", err))
+
 			_, err = metrics.GetFloat64Metric(gotMetrics, "host_uptime", map[string]string{}, false)
-			if err != nil {
-				ginkgo.Fail(fmt.Sprintf("Failed to find uptime metric: %v.\nHere is all NPD exported metrics: %v",
-					err, gotMetrics))
-			}
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to find uptime metric: %v.\nHere is all NPD exported metrics: %v", err, gotMetrics))
+		})
+
+		ginkgo.It("NPD should not report any problem", func() {
+			err := npd.WaitForNPD(instance, []string{"problem_gauge"}, 120)
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Expect NPD to become ready in 120s, but hit error: %v", err))
+
+			assertMetricValueInBound(instance,
+				"problem_gauge", map[string]string{"reason": "DockerHung", "type": "KernelDeadlock"},
+				0.0, 0.0)
+			assertMetricValueInBound(instance,
+				"problem_counter", map[string]string{"reason": "DockerHung"},
+				0.0, 0.0)
+			assertMetricValueInBound(instance,
+				"problem_counter", map[string]string{"reason": "FilesystemIsReadOnly"},
+				0.0, 0.0)
+			assertMetricValueInBound(instance,
+				"problem_counter", map[string]string{"reason": "KernelOops"},
+				0.0, 0.0)
+			assertMetricValueInBound(instance,
+				"problem_counter", map[string]string{"reason": "OOMKilling"},
+				0.0, 0.0)
+		})
+	})
+
+	ginkgo.Context("When ext4 filesystem error happens", func() {
+
+		ginkgo.BeforeEach(func() {
+			err := npd.WaitForNPD(instance, []string{"problem_gauge"}, 120)
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Expect NPD to become ready in 120s, but hit error: %v", err))
+			// This will trigger a ext4 error on the boot disk, causing the boot disk mounted as read-only and systemd-journald crashing.
```

Comment:
I am curious if this is actually making boot disk read-only and then crashing systemd-journald. Are you able to ssh and read metrics if that's real?

Reply:
> if this is actually making boot disk read-only and then crashing systemd-journald.

Yes :) It's a pretty significant failure, and is pretty dangerous. I think you have to reboot the node to recover from this.

> Are you able to ssh and read metrics if that's real?

Yes I can. SSH does not depend on a healthy disk, nor does NPD.